### DISCUSSION TITLE
What is the point of isError if there exists an error property in useQuery?

### DISCUSSION THREAD

**Discussion by lisherwin** [TanStack/query#6229] [Q&A] [answered]
What is the point of isError is when there is already an `error` property in the return object of `useQuery`? It seems like everywhere we use `isError`, we can simply check for the existence of the `error` object instead. In fact, the existence check e.g. `if (error)` is something we'd probably need to do downstream as well even if we checked `isError` to get the more granular error message or error type. My team has been thinking of removing all `isError` in our codebase, but we're not sure if there are unintended consequences or advantages we've missed from doing so.

Thanks in advance!

**Comment by TkDodo** [comment, marked as answer]
a query has 3 states: pending, success, error. If it's in error state (the booleans are just derived from that status), error can no longer be `null`. You can use that check or not, it's up to you

**Reply by lisherwin** [reply]
Thanks!
